src/main/java/net/minecraftforge/common/ForgeConfigSpec.java

```diff
@@ -905,13 +1033,23 @@ public void set(T value)
         public void clearCache() {
             this.cachedValue = null;
         }
+
+        public Supplier<ConfigGuiWidgetFactory> getScreenWidgetFactorySupplier() {
+            return this.screenWidgetFactorySupplier;
+        }
+
+        void setScreenWidgetFactorySupplier(final Supplier<ConfigGuiWidgetFactory> screenWidgetFactorySupplier)
+        {
+            this.screenWidgetFactorySupplier = screenWidgetFactorySupplier;
+        }
     }
 
     public static class BooleanValue extends ConfigValue<Boolean>
     {
         BooleanValue(Builder parent, List<String> path, Supplier<Boolean> defaultSupplier)
         {
             super(parent, path, defaultSupplier);
+            this.setScreenWidgetFactorySupplier(() -> ConfigGuiWidget.BooleanWidget.FACTORY);
```

Comment:
This and the others feel odd creating a new supplier for every single value instance, and instead should say have a `private static final Supplier<ConfigGuiWidgetFactory> FALLBACK_FACTORY = () -> factory` and pass that single instance fallback.

This way especially for large modpacks that may have tons of configs, we don't have the unnecessary overhead of every single config option creating a new supplier when they don't need to. (This sort of also is true regarding the default fallback of `() -> null`).